### ISSUE TITLE
[ResNet50/Pytorch] pytorch 1.7 support issue

### ISSUE THREAD

**Issue by gregmbi** [NVIDIA/DeepLearningExamples#776]
Related to **ConvNets classification** 

* *AMP* 
* *Pytorch 1.7*

Hi, do you plan to move this part of the repo to Pytorch 1.7 ? and start to use the internal apex ? i.e. the one integrated with pytorch. We tried running the current code with PyTorch 1.7 and see a strange issue, when running on an 8 GPU machine with distributed data parallel (as specified in the docs) there are 8 processes created for each GPU. 

We need to use features in pytorch 1.7, that is why we are asking.

**Comment by asulecki** [comment]
Hi @gregmbi,

There is a plan to migrate to PyTorch's native mixed precision implementation, unfortunatelly I can't provide a concrete deadline for it.

I think your issue is unrelated to PyTorch version or particular mixed precision implementation. Can you provide more details on your setup? 
First two questions that would help me help you are: 
 - Are you running through docker
 - Are you using SLURM or some other platform that automatically launches jobs on your machine, or do you have direct access to the machine?

**Comment by gregmbi** [comment]
We run through docker using the Nvidia 20.10 pytorch container. We are trying to migrate from the 20.6 container, the main difference being the move to pytorch 1.7.

We do not use Slurm

this is an example output from nvidia-smi:

```
Mon Dec  7 13:26:11 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.45.01    Driver Version: 455.45.01    CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  On   | 00000000:1D:00.0 Off |                  N/A |
| 53%   72C    P2   237W / 250W |   9720MiB / 11019MiB |     47%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 208...  On   | 00000000:1E:00.0 Off |                  N/A |
| 57%   77C    P2   237W / 250W |   9350MiB / 11019MiB |     84%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  GeForce RTX 208...  On   | 00000000:1F:00.0 Off |                  N/A |
| 56%   75C    P2   249W / 250W |   9350MiB / 11019MiB |     87%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  GeForce RTX 208...  On   | 00000000:20:00.0 Off |                  N/A |
| 52%   70C    P2   254W / 250W |   9350MiB / 11019MiB |     81%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   4  GeForce RTX 208...  On   | 00000000:21:00.0 Off |                  N/A |
| 53%   71C    P2   241W / 250W |   9350MiB / 11019MiB |     87%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   5  GeForce RTX 208...  On   | 00000000:22:00.0 Off |                  N/A |
| 58%   76C    P2   209W / 250W |   9350MiB / 11019MiB |     85%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   6  GeForce RTX 208...  On   | 00000000:23:00.0 Off |                  N/A |
| 58%   76C    P2   232W / 250W |   9350MiB / 11019MiB |     81%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   7  GeForce RTX 208...  On   | 00000000:24:00.0 Off |                  N/A |
| 53%   70C    P2   235W / 250W |   9350MiB / 11019MiB |     76%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A     12439      C   /opt/conda/bin/python3           9717MiB |
|    0   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    0   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    0   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    0   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    0   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    0   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    0   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12440      C   /opt/conda/bin/python3           9347MiB |
|    1   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    1   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12441      C   /opt/conda/bin/python3           9347MiB |
|    2   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    2   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12442      C   /opt/conda/bin/python3           9347MiB |
|    3   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    3   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12443      C   /opt/conda/bin/python3           9347MiB |
|    4   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    4   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12444      C   /opt/conda/bin/python3           9347MiB |
|    5   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    5   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    6   N/A  N/A     12476      C   /opt/conda/bin/python3           9347MiB |
|    6   N/A  N/A     12477      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12439      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12440      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12441      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12442      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12443      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12444      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12476      C   /opt/conda/bin/python3              0MiB |
|    7   N/A  N/A     12477      C   /opt/conda/bin/python3           9347MiB |
+-----------------------------------------------------------------------------+

```

**Comment by asulecki** [comment]
Could you share the exact command you use to launch container and training script?

**Comment by gregmbi** [comment]
python3 -m multiproc --nproc_per_node 8 main.py -c fanin --label-smoothing 0.1 --amp --static-loss-scale 256 --warmup 5 -j4 -p 50  /datasets/imagenet --data-backend pytorch --batch-size 128 --seed 0 --epochs 90 --lr-schedule step2 --arch resnet50 --lr 0.4

**Comment by gregmbi** [comment]
We also checked on our second machine with 8xv100 and got the same behaviour

**Comment by asulecki** [comment]
Recently the driver changed how it reports processes using CUDA. If you look closely on nvidia-smi output, you will see that there are only 8 processes, and each one is listed for every GPU, but it only uses one of the GPUs.

This is normal for newer driver versions, so you shouldnt be worried.

**Comment by gregmbi** [comment]
Really ? wow, i can say that the training seems to be working correctly but this is a really crappy output ... so much extra lines. Anyways thanks
